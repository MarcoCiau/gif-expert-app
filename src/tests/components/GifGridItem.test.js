import React, { Component } from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import GifGridItem from "../../components/GifGridItem";

describe("Test with <GifGridItem /> component", () => {
  const title = "my image";
  const url = "https://google.com";
  const wrapper = shallow(<GifGridItem title={title} url={url}></GifGridItem>);

  test("renders <GifGridItem /> correctly ", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("validate <GifGridItem /> cointains title prop in the <p></p> ", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("validate <GifGridItem /> cointains title & url prop in the <img></img> ", () => {
    const img = wrapper.find("img");
    expect(img.prop("src").trim()).toBe(url);
    expect(img.prop("alt").trim()).toBe(title);
  });

  test("validate <GifGridItem /> cointains classNames in the <div></div> ", () => {
    const div = wrapper.find("div");
    expect(div.hasClass("animate__fadeIn")).toBe(true);
  });
});
