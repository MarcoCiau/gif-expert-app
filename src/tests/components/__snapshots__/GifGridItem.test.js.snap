// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test with <GifGridItem /> component renders <GifGridItem /> correctly  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="my image"
    src="https://google.com"
  />
  <p>
    my image
  </p>
</div>
`;
